homeassistant/components/prusalink/__init__.py

```diff
@@ -63,30 +64,46 @@ class PrusaLinkUpdateCoordinator(DataUpdateCoordinator, Generic[T]):
     """Update coordinator for the printer."""
 
     config_entry: ConfigEntry
+    expect_change_until = 0.0
 
     def __init__(self, hass: HomeAssistant, api: PrusaLink) -> None:
         """Initialize the update coordinator."""
         self.api = api
 
         super().__init__(
-            hass, _LOGGER, name=DOMAIN, update_interval=timedelta(seconds=30)
+            hass, _LOGGER, name=DOMAIN, update_interval=self._get_update_interval(None)
         )
 
     async def _async_update_data(self) -> T:
         """Update the data."""
         try:
             with async_timeout.timeout(5):
-                return await self._fetch_data()
+                data = await self._fetch_data()
         except InvalidAuth:
             raise UpdateFailed("Invalid authentication") from None
         except PrusaLinkError as err:
             raise UpdateFailed(str(err)) from err
 
+        self.update_interval = self._get_update_interval(data)
+        return data
+
     @abstractmethod
     async def _fetch_data(self) -> T:
         """Fetch the actual data."""
         raise NotImplementedError
 
+    @callback
+    def expect_change(self) -> None:
+        """Expect a change."""
+        self.expect_change_until = monotonic() + 30
+
+    def _get_update_interval(self, data: T) -> timedelta:
+        """Get new update interval."""
+        if self.expect_change_until > monotonic():
+            return timedelta(seconds=3)
```

Comment:
Normally we don't ever poll faster than 5 seconds.

Reply:
You're right. This is in short bursts. Their own frontend does every 2 seconds but that didn't feel right. 
```suggestion
            return timedelta(seconds=5)
```